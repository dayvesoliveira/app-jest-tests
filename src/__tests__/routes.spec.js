import React from 'react'
import { shallow } from 'enzyme'
import { MemoryRouter } from 'react-router'
import Routes from '../routes'

describe('routes using memory router', () => {

    it('should show Home component for / router (using memory router)', () => {
        const component = shallow(
            <MemoryRouter initialEntries={['/']} >
                <Routes />
            </MemoryRouter>
        )
        expect(component).toMatchSnapshot()
        expect(component.find(Routes)).toHaveLength(1)
    })

})