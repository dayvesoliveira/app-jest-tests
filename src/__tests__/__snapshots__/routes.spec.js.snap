// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes using memory router should show Home component for / router (using memory router) 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "faek8l",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "faek8l",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Routes />
</Router>
`;
